### ISSUE TITLE
annoying console errors about "undefined is not a valid URL"

### ISSUE THREAD

**Issue by tiansh** [EFForg/https-everywhere#18020]
Type: code issue

Maybe similar to #16562. If there is a web request from another webextension's content page, an error with message "undefined is not a valid URL" will appear in the console.

Stack information for the Error is:

> undefined is not a valid URL.
>     /background-scripts/background.js:321

```javascript
    if ('documentUrl' in details) { // Firefox 54+
      firstPartyHost = new URL(details.documentUrl).host; // documentUrl is undefined here
    } else if ('originUrl' in details) { // Firefox < 54
      firstPartyHost = new URL(details.originUrl).host;
    } else if('initiator' in details) { // Chrome
      firstPartyHost = new URL(details.initiator).host;
    }
```

## Reproduce:

1. Create a clean Firefox profile, launch Firefox
2. Install HTTPS Everywhere from AMO
3. Download https://gist.github.com/tiansh/59855bd267ba35db4499e8114ab789a1 and install it as temp extension (debug); This extension may be replaced by any extensions try to request resource from content script
4. Open debug panel for HTTPS Everywhere 
5. Open https://www.example.com/

**Comment by pipboy96** [comment]
I also noticed that before but didn't bother to create an issue.

**Comment by cschanaj** [comment]
Closed by #19636
